src/HasBinaryUuid.php

```diff
@@ -73,7 +73,54 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        return array_merge(parent::toArray(), [$this->getKeyName() => $this->uuid_text]);
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+        foreach ($uuidAttributes as $attributeKey) {
+            if (array_key_exists($attributeKey, $array)) {
+                $array[$attributeKey] = $this->{"{$attributeKey}_text"};
```

Comment:
It would be nice if we shouldn't have to hard code the `_text` naming..

Reply:
Yes true... i agree with @brendt